public/app/features/logs/logsFrame.ts

```diff
@@ -103,9 +103,15 @@ export function parseDataplaneLogsFrame(frame: DataFrame): LogsFrame | null {
 }
 
 export function parseLogsFrame(frame: DataFrame): LogsFrame | null {
-  if (frame.meta?.type === DataFrameType.LogLines) {
-    return parseDataplaneLogsFrame(frame);
-  } else {
-    return parseLegacyLogsFrame(frame);
+  frame.meta = frame.meta ?? {};
+  frame.meta.custom = frame.meta.custom ?? {};
+  if (frame.meta.custom.logsFrame == null) {
```

Comment:
bringing our slack discussion into the PR: 

There is some regression risk here if the dataframe passed in is mutated and then the logs frame is generated from that mutated dataframe, we'd carry the cached logs frame labels, and the timestamp/body/extra fields into the generated LogsFrame which could introduce regression.

Although the risk of that is theoretical, and probably low enough to warrant merging as is (plus we should fix any unexpected mutations like this), but it might be worth at least checking the config and labels for changes to invalidate the cache if something was mutated?

Reply:
776ae918465bc2f403d898a1409b63fb00a5944e